### PR TITLE
Expand the matching files criteria in the protocol import transformat…

### PR DESCRIPTION
**Pull request by JorMaister**
…ion matrix so the tsId or the normalized basename are checked.

### CODE REVIEW

File: imod/protocols/protocol_importSetOfTM.py [hunk 1]
```diff
@@ -187,8 +190,8 @@ def _validate(self):
         if matchingFiles:
             tsIdList = self.getInputSet().getTSIds()
             tmFileList = [normalizeTSId(fn) for fn, _ in self.iterFiles()]
-            self.matchingFiles = list(set(tsIdList) & set(tmFileList))
-            if not self.matchingFiles:
+            self.matchingTsIds = list(set(tsIdList) & set(tmFileList))
+            if not self.matchingTsIds:
```
Comment (azazellochg) [inline imod/protocols/protocol_importSetOfTM.py:194]: I assume we don't want to throw an error anymore? Then self.matchingTsIds can be defined in the self._initialize()

File: imod/protocols/protocol_importSetOfTM.py [hunk 2]
```diff
@@ -106,10 +106,13 @@ def generateTransformFileStep(self, tsId, matchBinningFactor):
...
         outputTransformFile = self.getExtraOutFile(tsId, ext=XF_EXT)
-        self.debug(f"Matching files: {self.matchingFiles}")
+        self.debug(f"Matching tsIds: {self.matchingTsIds}")
 
         for tmFilePath, _ in self.iterFiles():
-            if pwutils.removeBaseExt(tmFilePath) in self.matchingFiles:
+            tmFileBaseName = pwutils.removeBaseExt(tmFilePath)
+            # Look for basename - tsId or base -name normalized basename - tsId matches. See tomo.convert.mdoc
+            # normalizeTSId
+            if tmFileBaseName in self.matchingTsIds or normalizeTSId(tmFileBaseName) in self.matchingTsIds:
```
Comment (azazellochg) [inline imod/protocols/protocol_importSetOfTM.py:115]: isn't it the opposite, e.g. look for TSID in the filename: TS_28 in mytiltseries_TS_28_lalala ?

File: imod/protocols/protocol_importSetOfTM.py (hunk 1 above)
Comment (JorMaister) [reply imod/protocols/protocol_importSetOfTM.py:194]: In the end, in that part I only renamed the matchingFiles class attribute by matchingTsIds for clarity. But I can undo that part. It does not affect the name matching.

File: imod/protocols/protocol_importSetOfTM.py (hunk 2 above)
Comment (JorMaister) [reply imod/protocols/protocol_importSetOfTM.py:115]: Sure, it's more robust that way. I was dealing as a reference with numerical TS names to which TS_ is added. We may add the {TS} pattern to the imports to offer more versatility and make it easier for the users to make the matching when importing files that way, avoiding possible manual renaming.

File: imod/protocols/protocol_importSetOfTM.py (hunk 1 above)
Comment (azazellochg) [reply imod/protocols/protocol_importSetOfTM.py:194]: Ok, I need to read the whole code properly. I'll do it later tonight.

Comment (JorMaister) [reply imod/protocols/protocol_importSetOfTM.py:194]: No worries. Many thanks, Grigory.

**Review by azazellochg — APPROVED** [review]

**Pull request merged**